packages/assets-controllers/src/TokenRatesController.ts

```diff
@@ -262,9 +247,6 @@ export class TokenRatesController extends StaticIntervalPollingControllerV1<
       ) {
         this.update({ contractExchangeRates: {} });
         this.configure({ chainId, nativeCurrency: ticker });
-        if (this.#pollState === PollState.Active) {
-          await this.updateExchangeRates();
```

Comment:
I'm unsure about the network switch pattern.  One pattern is like this, where the controller handles `onNetworkStateChange` within itself.  Another pattern is how extension handles `networkDidChange` from outside the controller e.g. [for currency rate.](https://github.com/MetaMask/metamask-extension/blob/d143cbfd4ce3139843c9c298f2141c3b35139dd5/app/scripts/metamask-controller.js#L1631
)

It seems like the latter is better for the multichain polling approach?  Since client is the one calling `startPollingByNetworkClientId`, it would also have to do the stopping / restarting.  Therefore this trigger is migrated to client.